### PR TITLE
Break function out for database resets in tests

### PR DESCRIPTION
**Pull request by squirrelo**
~~Requires #430~~

This moves most of the functions that previously resided in the `ag` script to the `env_management.py` file. This allows changes added to the test checker decorator that reset the database after each test. 

This is a major push for #434 (adding travis testing)

### CODE REVIEW

**Comment by adamrp** [comment]
@squirrelo this seems to contain commits that were merged in other recent PRs (e.g., #429, #430). Is there a reason for the duplication? Maybe you can pull upstream master to get around it?

**Comment by squirrelo** [comment]
Yeah sorry, juggling many pull requests. Done.

**Comment by adamrp** [comment]
No problem! Thanks :smile:

File: amgut/lib/data_access/env_management.py [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+
+    # Get the cursor
+    cur = conn.cursor()
+
+    cur.execute('SET SEARCH_PATH TO ag, barcodes, public')
+
+    db_exists = _check_db_exists(AMGUT_CONFIG.database, cur)
+
+    # Check that the database does not already exist
+    if db_exists and force:
```
Comment (adamrp) [inline amgut/lib/data_access/env_management.py]: I find this construction just slightly confusing, and would find a nested `if` clearer:
```python
if db_exists:
    if force:
        return
    raise ValueError("Database '{}' already present on the system".format(AMGUT_CONFIG.database))
```

**Comment by adamrp** [comment]
Besides minor comment on clarity (not blocking), :+1:

**Pull request merged**